src/runtime_src/core/common/sensor.cpp

```diff
@@ -167,9 +168,12 @@ read_data_driven_electrical(const std::vector<xq::sdm_sensor_info::data_type>& c
      * So, actual sensor value => 12000 * 10 ^ (-3) = 12 Volts.
      */
     auto unitm = pow(10, tmp.unitm);
-    pt.put("voltage.volts", static_cast<double>(tmp.input) * unitm);
-    pt.put("voltage.max", static_cast<double>(tmp.max) * unitm);
-    pt.put("voltage.average", static_cast<double>(tmp.average) * unitm);
+    stream << std::fixed << std::setprecision(3) << static_cast<double>(tmp.input) * unitm;
+    pt.put("voltage.volts", stream.str());
+    stream << std::fixed << std::setprecision(3) << static_cast<double>(tmp.max) * unitm;
```

Comment:
Don't you need to reset the stream? It gets concatenated to existing stream object.

Reply:
taken care of this. Thank you

Reply:
Better to write simple function to get the value. I see the same line in 6 different places.

Reply:
In format_base10_shiftdown3() API, decimal_shift value is 1000 and it is constant. In our case, this value may not be 1000 and this value is derived from unitm for us.
So, I can't directly use this function here.

Reply:
As discussed, i was asking you to create one more generic function. Latest changes looks good